src/app/acorns/acorns.controller.js

```diff
@@ -5,56 +5,53 @@
  * Created by vytautassugintas on 06/04/16.
  */
 (function () {
-    'use strict';
-    var AcornController = function ($http, $scope, $window, $cookies, $timeout, $httpParamSerializer, $filter, ProfileService, Challenges, Resources) {
-
-        $scope.greeting = "hello";
-
-        $scope.showLoader = true;
-
-        $scope.userAvailableKudos = 0;
-        $scope.buttonDisabled = true;
-
-        $scope.receivedAcorns = false;
-        $scope.sentAcorns = true;
-
-        $scope.isValid = isValid;
-        $scope.checkUser = checkUser;
-
-        checkUser();
-
-        ProfileService.remainingKudos().then(function (val) {
-            Resources.setUserAvailableKudos(val);
-        });
-
-        ProfileService.receivedKudos().then(function (val) {
-            $scope.userReceivedKudos = val;
-            $scope.showLoader = false;
-        });
-
-        $scope.$watch(function () {
-            return Resources.getUserAvailableKudos()
-        }, function (newVal) {
-            if (!isValid(newVal)) $scope.userAvailableKudos = Resources.getUserAvailableKudos();
-        });
-
-        function isValid(value) {
-            return typeof value === "undefined";
-        }
-
-        function checkUser() {
-            ProfileService.checkUser().then(function (val) {
-                val.logged ? $window.location.href = "#/acorns" : $window.location.href = "#/login";
+        'use strict';
+        var AcornController = function ($http, $scope, $window, $cookies, $timeout, $httpParamSerializer, $filter, ProfileService, Challenges, Resources) {
+            $scope.showLoader = true;
```

Comment:
Rule 7 in JohnPapas Styleguide - https://github.com/johnpapa/angular-styleguide/blob/master/a1/README.md#style-y030

Try to use controller as, instead of $scope as much as possible. ControllerAs makes scope pollution much less of a problem and helps you nest, access and control lots of scopes.